### DISCUSSION TITLE
ImageMagick background check

### DISCUSSION THREAD

**Discussion by Ant4187** [ImageMagick/ImageMagick#3938] [Digital Image Processing]
I could do with finding out whether a transparent PNG image has a purely white (#fffff) foreground using ImageMagick/imagick.

If any pixels are not #ffffff or transparent, the image would need to be rejected.

Any help would be appreciated, I'm not really sure how to go about this.

**Comment by snibgo** [comment]
What version of IM, on what platform?

Every pixel in your image should either be opaque white, or fully transparent, with no partial transparency. Is that correct? If so, then flatten against a white background, and test whether any pixels are not white, ie whether the minimum value is white.
